src/main/java/Factory.java

```diff
@@ -14,13 +15,12 @@
     static ArrayList<Integer> incomingList = new ArrayList();
     static ArrayList<Integer> outgoingList = new ArrayList();
```

Comment:
http://qaru.site/questions/51268/polymorphism-why-use-list-list-new-arraylist-instead-of-arraylist-list-new-arraylist